extensions/amp-ad/0.1/amp-ad.js

```diff
@@ -69,10 +69,12 @@ export class AmpAd extends AMP.BaseElement {
 
       // TODO(tdrl): Check amp-ad registry to see if they have this already.
       if (!a4aRegistry[type] ||
-          !a4aRegistry[type](this.win, this.element)) {
-        // Network either has not provided any A4A implementation or the
-        // implementation exists, but has explicitly chosen not to handle this
-        // tag as A4A.  Fall back to the 3p implementation.
+          !a4aRegistry[type](this.win, this.element) ||
+          this.win.document.querySelector('meta[name=amp-3p-iframe-src]')) {
```

Comment:
This check needs to be done higher, so the [assert](https://github.com/ampproject/amphtml/pull/7906/files#diff-03f204ef56e45e6c843922d48e709848R58) can be updated.